### ISSUE TITLE
add documentation for event listeners

### ISSUE THREAD

**Issue by siqnole** [EnokiUN/voltage#61]
### Summary

adding documentation for event listeners and what they listen for

### What is the feature request for?

The documentation

### The Problem

not annoying enoki when i dont know how to do something because of the lack of documentation on the subject

### The Ideal Solution

adding documentation going into a short detail on how to listen for message edits, deletes, or sends and using them to create commands

### The Current Solution

annoying enoki

**Comment by 0x454d505459** [comment]
I agree, that could be usefull. Because I don't know what are the possible events to use. The events name from the [revolt.js](https://revolt.js.org/classes/Client.Client-1.html#on) page doesn't seems to work.

**Comment by EnokiUN** [comment]
well atleast untill readthedocs stops being weird ~~or I stop being stupid~~ all the possible events can be found by going to `internals/ws.py` and searching for `dispatch`

**Comment by 0x454d505459** [comment]
Thanks, that is usefull. Gonna list all of them:
- onReady : "ready" 
- onMessage : "message"
- onMessageEdit : "message_update"
- onmessageDelete : "message_delete"
- onChannelCreate : "channel_create"
- oncChannelEdit : "channel_update"
- onChannelDelete : "channel_delete"
- onGroupChannelJoin : "group_channel_join"
- onGroupChannelLeave : "group_channel_leave"
- onUserStartsTyping : "channel_start_typing"
- onUserStopsTyping : "channel_stop_typing"
- onServerEdit : "server_update"
- onServerDelete : "server_delete"
- onServerMemberEdit : "server_member_update"
- onMemberJoin : "member_join"
- onMemberLeave : "member_leave"
- onRoleEdit : "server_role_update"
- onRoleDelete : "server_role_delete"
- onUserEdit : "user_update"

That's all of it

**Comment by ksIsCute** [comment]
> well atleast untill readthedocs stops being weird ~or I stop being stupid~ all the possible events can be found by going to `internals/ws.py` and searching for `dispatch`

im as clueless as u are with the docs situation at this point it would be in our best interest to kind of factory reset the docs to see why they're failing

**Comment by 0x454d505459** [comment]
Could be a solution

**Comment by ksIsCute** [comment]
> Could be a solution

best to try it if enoki ever gets online today

**Comment by EnokiUN** [comment]
> > well atleast untill readthedocs stops being weird ~or I stop being stupid~ all the possible events can be found by going to `internals/ws.py` and searching for `dispatch`
> 
> im as clueless as u are with the docs situation at this point it would be in our best interest to kind of factory reset the docs to see why they're failing

I've tried, I really have

**Comment by ksIsCute** [comment]
> > > well atleast untill readthedocs stops being weird ~or I stop being stupid~ all the possible events can be found by going to `internals/ws.py` and searching for `dispatch`
> > 
> > 
> > im as clueless as u are with the docs situation at this point it would be in our best interest to kind of factory reset the docs to see why they're failing
> 
> I've tried, I really have

oh i know you've basically gone insane over the docs problem, at this point would it be in our best interest to just, make it? instead of using readthedocs.io, we can just make a website for it? it seems like it would be a quick fix, or maybe a permanent one? because we can basically have all the same items and more detail if we need it. plus no more rst

**Comment by EnokiUN** [comment]
> > well atleast untill readthedocs stops being weird ~or I stop being stupid~ all the possible events can be found by going to `internals/ws.py` and searching for `dispatch`
> 
> im as clueless as u are with the docs situation at this point it would be in our best interest to kind of factory reset the docs to see why they're failing

I've tried, I really have

> > > > well atleast untill readthedocs stops being weird ~or I stop being stupid~ all the possible events can be found by going to `internals/ws.py` and searching for `dispatch`
> > > 
> > > 
> > > im as clueless as u are with the docs situation at this point it would be in our best interest to kind of factory reset the docs to see why they're failing
> > 
> > 
> > I've tried, I really have
> 
> oh i know you've basically gone insane over the docs problem, at this point would it be in our best interest to just, make it? instead of using readthedocs.io, we can just make a website for it? it seems like it would be a quick fix, or maybe a permanent one? because we can basically have all the same items and more detail if we need it. plus no more rst

I'm still going to use rst as it builds without any issue, even on my phone.

I've considered using a website but that requires getting a domain and hosting it.

Maybe github pages will bring us salvation?

**Comment by ksIsCute** [comment]
github pages is perfect, because we can just host the pages site in a folder unrelated to voltage, plus if u need css or just help with starting and styling the site :trollface: ask me

**Comment by 0x454d505459** [comment]
We could just use markdown, also github as a wiki feature

**Comment by ksIsCute** [comment]
yeah but that limits our options and our ability to style certain objects and to display and optimise

**Comment by ksIsCute** [comment]
we could use markdown, html, and rst, but html and rst are probably our best bet if we want the site to be "bearable"

**Comment by EnokiUN** [comment]
Issue isn't rst, infact it's making life easier. Issue is rtd not working with attributetable for some reason SPECIFICALLY for Voltage.

Maybe it's something stupid I did. idk

**Comment by ksIsCute** [comment]
lets just use gh pages with rst then, because without docs that will drive people away and also "limit" abilities (even thougbh they're a thing, they arent in the documentation, leaving people clueless)

**Comment by 0x454d505459** [comment]
We could just setup a github pages running [facebook's docusaurus](https://github.com/facebook/docusaurus)

**Comment by ksIsCute** [comment]
good idea

**Comment by EnokiUN** [comment]
Let me try a thing

**Comment by ksIsCute** [comment]
make a new branch and start gh pages in there

**Comment by EnokiUN** [comment]
I started a gh page and uhhhh

css died?

**Comment by ksIsCute** [comment]
wat

**Comment by EnokiUN** [comment]
https://enokiun.github.io/voltage l

**Comment by ksIsCute** [comment]
i think we should start from scratch

**Comment by EnokiUN** [comment]
Both you and I know damn well that I no longer have the physical or mental requirements for that especially now that I'm coding on phone

**Comment by EnokiUN** [comment]
additionally, Leave this for the actual issue at hand, move to the docs discussion

**Comment by EnokiUN** [comment]
if someone is willing to, go to `docs/source/api.rst` and add a list of event names and the args they pass

**Comment by EnokiUN** [comment]
to generate the docs you cd into `docs` and run `make html` tho rtd does that automatically for every commit

**Comment by ksIsCute** [comment]
🇧🇳

**Comment by ksIsCute** [comment]
> if someone is willing to, go to `docs/source/api.rst` and add a list of event names and the args they pass

got  it
